spec/replication/util/api_attributes_spec.rb

```diff
@@ -0,0 +1,33 @@
+describe "all tables" do
+  let(:connection) { ApplicationRecord.connection }
+
+  it "do not have an attribute called href" do
+    href_tables = []
+    connection.tables.each do |t|
+      next if %w(schema_migrations ar_internal_metadata).include?(t)
+      href_tables << t if connection.columns(t).collect(&:name).include?("href")
+    end
+    expect(href_tables.size).to eq(0), <<-EOS.gsub!(/^ +/, "")
+      Attribute "href" was found for the following table(s):
+
+      #{href_tables.join("\n")}
+
+      The "href" attribute is reserved for the ManageIQ API.
+    EOS
```

Comment:
I recommend that you create a method out of the error message, so that it doesn't "double display" in rspec errors.  It's hard to explain, but the way rspec displays errors, it show the line of code in error followed by the error message.  Because the line of code has the message as a string, you see it "twice" in the output.  The first is non-interpolated as it's the source code, and the second it interpolated.  However, it's extremely confusing to see in the output.  By pushing the error message building into a method like `error_message_for`, then when you see the output, you don't see the non-interpolated source based version.

Hope that makes sense...we did something similar here: https://github.com/ManageIQ/manageiq/blob/master/spec/replication/util/migration_order_spec.rb#L20  (we still have to do this for the other tests in this directory, but that particular one is more common to fail and is why we did it first).